MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/util/Login.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.macrobenchmark.target.util
+
+abstract class Login {
+    abstract suspend fun isLoggedIn(): Boolean
+    abstract suspend fun login(): Boolean
+    abstract suspend fun logout()
+
+    lateinit var password: String
+
+    lateinit var userName: String
```

Comment:
Is it WIP? 


1. This is not set anywhere and therefore any method called from this class will crash.
1. If you want to have these as params, I'd suggest using `String?` instead of `lateinit var`, because you can easily check for nullability and prevent a crash. With lateinit you can check for initialized, but the caller has no idea it may not be set

1. I also think this is implementation detail of the `Login(Usecase?)` and shouldn't be exposed .. maybe just username

Reply:
Yeah, that was WIP. Now updated.